clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -37,8 +37,10 @@
     private ConnectionState connectionState = null;
     private HttpClient httpClient;
     private String stopError;
+    private CompletableFuture<Void> handshakeResponseFuture;
+    private Duration handshakeResponseTimeout = Duration.ofSeconds(15);
 
-    HubConnection(String url, Transport transport, boolean skipNegotiate, Logger logger, HttpClient httpClient, Supplier<CompletableFuture<String>> accessTokenProvider) {
+    HubConnection(String url, Transport transport, boolean skipNegotiate, Logger logger, HttpClient httpClient, Supplier<CompletableFuture<String>> accessTokenProvider, Duration handshakeResponseTimeout) {
```

Comment:
Not something we need to do right now since it's all internal, but I do think maybe keeping `HttpConnectionOptions` around as an internal carrier might be good :trollface:

Reply:
*facepalm

Reply:
Not to say I told you so but.... 😄

Reply:
Pretty sure [I also told me so](https://github.com/aspnet/SignalR/pull/3078#issuecomment-427954726) :)